client/scripts/controllers/server/threads.ts

```diff
@@ -456,6 +456,26 @@ class ThreadsController {
     });
   }
 
+  public async fetchThreadForSnapshot(args: { snapshot: string }) {
+    const response = await $.ajax({
+      url: `${app.serverUrl()}/fetchThreadForSnapshot`,
+      type: 'GET',
+      data: {
+        snapshot: args.snapshot,
+        chain: app.activeChainId(),
+      },
+    });
+    if (response.status !== 'Success') {
+      throw new Error(`Cannot fetch thread: ${response.status}`);
+    }
+
+    const thread = modelFromServer(response.result);
```

Comment:
I recommend having this new route return an id or link rather than the entire thread object.